### PR TITLE
Add error prop to generated vcs compile script

### LINKED ISSUE #163: VCS compile script doesn't propagate errors
Bender's script output in the vcs tera template emits a bunch separate parsing commands. Any individual command may fail but if the last command succeed the script is observed to succeed instead of stopping on an error.

### PR DESCRIPTION
**Pull request by uge**
Change the VCS compile script template to start with a 
```set -e```
to ensure error propagation to fix #163.

### CODE REVIEW

File: src/script_fmt/vcs_sh.tera [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 # {{ HEADER_AUTOGEN }}
+
+# Set propagation of error to exit on first error
+set -e
```
Comment (micprog) [inline src/script_fmt/vcs_sh.tera]: I think masking this behind the existing `--no-abort-on-error` script option makes sense, for consistency.

```suggestion
{% if abort_on_error %}# Set propagation of error to exit on first error
set -e
{% endif %}
```

**Review by micprog — CHANGES_REQUESTED** [review]

**Comment by uge** [comment]
I wasn't familiar with that flag - thanks for the suggestion.

**Review by micprog — APPROVED** [review]
LGTM! :+1:

**Review by micprog — APPROVED** [review]
LGTM :+1:

**Pull request merged**